app/dashboard/notifications.py

```diff
@@ -544,43 +520,56 @@ def maybe_market_to_github(bounty, event_name, profile_pairs=None):
 
     # Prepare the comment message string.
     msg = build_github_notification(bounty, event_name, profile_pairs)
+    if msg == bounty.done_comment:
+            delete_issue_comment(bounty.done_comment, username, repo)
+
     if not msg:
         return False
 
+    #try either posting or patching the comment on the github issue
     try:
         username = uri_array[1]
         repo = uri_array[2]
         issue_num = uri_array[4]
 
-        if event_name == 'work_started':
-            comment_id = bounty.interested_comment
-        elif event_name in ['work_done', 'work_submitted']:
-            comment_id = bounty.submissions_comment
-
-        # Handle creating or updating comments if profiles are provided.
-        if event_name in ['work_started', 'work_submitted'] and profile_pairs:
-            if comment_id is not None:
-                patch_issue_comment(comment_id, username, repo, msg)
+        if event_name in ['new_bounty', 'increased_bounty']:
+            if bounty.opened_comment:
+                patch_issue_comment(bounty.opened_comment, username, repo, msg)
             else:
                 response = post_issue_comment(username, repo, issue_num, msg)
-                if response.get('id'):
-                    if event_name == 'work_started':
-                        bounty.interested_comment = int(response.get('id'))
-                    elif event_name in ['work_done', 'work_submitted']:
-                        bounty.submissions_comment = int(response.get('id'))
-                    bounty.save()
-        # Handle deleting comments if no profiles are provided.
-        elif event_name in ['work_started'] and not profile_pairs:
-            if comment_id:
-                delete_issue_comment(comment_id, username, repo)
-                if event_name == 'work_started':
-                    bounty.interested_comment = None
-                elif event_name == 'work_done':
-                    bounty.submissions_comment = None
-                bounty.save()
-        # If this isn't work_started/done, simply post the issue comment.
-        else:
-            post_issue_comment(username, repo, issue_num, msg)
+                bounty.opened_comment = int(response.get('id'))
+        elif event_name == 'work_started':
+            if bounty.interested.filter(pending=True).count():
+                if bounty.interested_comment:
```

Comment:
You can reuse the existing comment field for all of these patches.  Also can we DRY this up?

Reply:
@mbeacom Done!